x-pack/plugins/alerting/server/lib/is_rule_snoozed.ts

```diff
@@ -6,47 +6,60 @@
+export function getActiveSnoozes(
+  rule: RuleSnoozeProps
+): Array<{ snoozeEndTime: Date; id: string }> | null {
   if (rule.snoozeSchedule == null) {
     return null;
   }
 
   const now = Date.now();
-  for (const snooze of rule.snoozeSchedule) {
-    const { duration, rRule } = snooze;
-    const startTimeMS = Date.parse(rRule.dtstart);
-    const initialEndTime = startTimeMS + duration;
-    // If now is during the first occurrence of the snooze
-
-    if (now >= startTimeMS && now < initialEndTime) return new Date(initialEndTime);
-
-    // Check to see if now is during a recurrence of the snooze
-    if (rRule) {
-      try {
-        const rRuleOptions = {
-          ...rRule,
-          dtstart: new Date(rRule.dtstart),
-          until: rRule.until ? new Date(rRule.until) : null,
-          wkst: rRule.wkst ? Weekday.fromStr(rRule.wkst) : null,
-          byweekday: rRule.byweekday ? parseByWeekday(rRule.byweekday) : null,
+  return rule.snoozeSchedule
+    .map((snooze) => {
+      const { duration, rRule, id } = snooze;
+      const startTimeMS = Date.parse(rRule.dtstart);
+      const initialEndTime = startTimeMS + duration;
+      // If now is during the first occurrence of the snooze
+
+      if (now >= startTimeMS && now < initialEndTime)
+        return {
+          snoozeEndTime: new Date(initialEndTime),
+          id,
         };
 
-        const recurrenceRule = new RRule(rRuleOptions);
-        const lastOccurrence = recurrenceRule.before(new Date(now), true);
-        if (!lastOccurrence) continue;
-        const lastOccurrenceEndTime = lastOccurrence.getTime() + duration;
-        if (now < lastOccurrenceEndTime) return new Date(lastOccurrenceEndTime);
-      } catch (e) {
-        throw new Error(`Failed to process RRule ${rRule}: ${e}`);
+      // Check to see if now is during a recurrence of the snooze
+      if (rRule) {
+        try {
+          const rRuleOptions = {
+            ...rRule,
+            dtstart: new Date(rRule.dtstart),
+            until: rRule.until ? new Date(rRule.until) : null,
+            wkst: rRule.wkst ? Weekday.fromStr(rRule.wkst) : null,
+            byweekday: rRule.byweekday ? parseByWeekday(rRule.byweekday) : null,
+          };
+
+          const recurrenceRule = new RRule(rRuleOptions);
+          const lastOccurrence = recurrenceRule.before(new Date(now), true);
+          if (!lastOccurrence) return null;
+          const lastOccurrenceEndTime = lastOccurrence.getTime() + duration;
+          if (now < lastOccurrenceEndTime)
+            return { snoozeEndTime: new Date(lastOccurrenceEndTime), id };
+        } catch (e) {
+          throw new Error(`Failed to process RRule ${rRule}: ${e}`);
+        }
       }
-    }
-  }
+      return null;
+    })
+    .filter(Boolean) as Array<{ snoozeEndTime: Date; id: string }>;
+}
 
-  return null;
+export function getRuleSnoozeEndTime(rule: RuleSnoozeProps): Date | null {
+  return first(getActiveSnoozes(rule))?.snoozeEndTime ?? null;
```

Comment:
Question, do we need to sort it by snoozeEndTime first before getting the first item.